samtranslator/translator/arn_generator.py

```diff
@@ -42,7 +48,17 @@ def get_partition_name(cls, region=None):
         if region is None:
             # Use Boto3 to get the region where code is running. This uses Boto's regular region resolution
             # mechanism, starting from AWS_DEFAULT_REGION environment variable.
-            region = boto3.session.Session().region_name
+
+            if ArnGenerator.class_boto_session is None:
+                region = boto3.session.Session().region_name
+            else:
+                region = ArnGenerator.class_boto_session.region_name
+
+        # If region is still None, then we could not find the region. This will only happen
+        # in the local context. When this is deployed, we will be able to find the region like
+        # we did before.
+        if region is None:
+            raise NoRegionFound("AWS Region cannot be found")
```

Comment:
We could maybe add `if running locally, is AWS_DEFAULT_REGION set?` at the end of the message?

Reply:
Customer's shouldn't have to set an Env Var. With this change, SAM will accept a Boto Session. Which then SAM can used to create clients with those creds. SAM CLI will then pass this information to SAM, which will resolve the failures SAM CLI sees.